File: Edward Poot/src/ql/ast/expr/binaryoperatorexpr/lt.go
package binaryoperatorexpr

import (
	"ql/ast/expr"
	"ql/ast/visit"
)

type LT struct {
	Lhs, Rhs expr.Expr
}

func (l LT) GetLhs() expr.Expr {
	return l.Lhs
}

func (l LT) GetRhs() expr.Expr {
	return l.Rhs
}

func (l LT) Eval() interface{} {
	return l.GetLhs().Eval().(int) < l.GetRhs().Eval().(int)
}

func (l LT) Accept(v visit.Visitor, s interface{}) interface{} {
	return v.Visit(l, s)
}

File: Kevin van den Bekerom/DSLQL/src/main/java/org/uva/sea/ql/ast/expr/LT.java
package org.uva.sea.ql.ast.expr;

import org.uva.sea.ql.ast.form.Context;
import org.uva.sea.ql.ast.visit.Visitable;
import org.uva.sea.ql.ast.visit.Visitor;

public class LT extends BinaryExpr implements Visitable {
	
	public LT(Expr lhs, Expr rhs) {
		super.lhs = lhs;
		super.rhs = rhs;
	}
	
	@Override
	public Boolean eval() {
		return (Integer) lhs.eval() < (Integer) rhs.eval();
	}
	
	public void accept(Visitor visitor) {
		visitor.visit(this, null);
	}

	@Override
	public String toString() {
		return super.toString() + " (<)";
	}
	
	@Override
	public Type getType(Context context) {
		return Type.BOOLEAN;
	}
}
